### PR TITLE
Added SDK examples

### PR DESCRIPTION
**Pull request by vidhutsingh**
Descriptions :
Added SDK examples. 

Signed-off-by: vidhut-kumar-singh <vidhut-kumar.singh@hpe.com>

### CODE REVIEW

File: examples/fields.go [hunk 1]
```diff
@@ -0,0 +1,70 @@
+package main
```
Comment (rgcostea) [inline examples/fields.go:2]: missing copyright

Comment (vidhutsingh) [reply examples/fields.go:2]: thanks @rgcostea

File: examples/fields.go [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+	"github.com/hpe-storage/nimble-golang-sdk/pkg/client/v1/nimbleos"
+	"github.com/hpe-storage/nimble-golang-sdk/pkg/param"
+	"github.com/hpe-storage/nimble-golang-sdk/pkg/service"
+)
+
+func main() {
+	groupService, err := service.NewNsGroupService("1.1.1.1", "xxx", "xxx", "v1", true)
+	if err != nil {
+		fmt.Printf("NewGroupService(): Unable to connect to group, err: %v", err.Error())
```
Comment (rkumpf) [inline examples/fields.go:15]: Is there any point in proceeding, or should it just exit?

File: examples/fields.go [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+		ThinlyProvisioned: param.NewBool(true),
+		Online:            param.NewBool(true),
+		LimitIops:         &limitIopsField,
+		LimitMbps:         &limitMbpsField,
+		MultiInitiator:    param.NewBool(true),
+		AgentType:         nimbleos.NsAgentTypeNone,
+	}
+
+	// create volume
+	volume, err := volSvc.CreateVolume(newVolume)
```
Comment (rkumpf) [inline examples/fields.go:42]: Include error checks?

File: examples/fields.go [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+	// Initialize volume attributes
+	var sizeField int64 = 5120
+	descriptionField := "This volume was created as part of a unit test"
+	var limitIopsField int64 = 256
+	var limitMbpsField int64 = 1
+
+	// set volume attributes
+	newVolume := &nimbleos.Volume{
+		Name:              param.NewString("TestDemo"),
+		Size:              &sizeField,
```
Comment (rkumpf) [inline examples/fields.go:31]: Why not use param.NewInt64()?

File: examples/filters.go [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+	// set debug
+	groupService.SetDebug()
+
+	// get volume service instance
+	volSvc := groupService.GetVolumeService()
+
+	// init param
+	sfilter := &param.GetParams{}
+	// set search filter
+	f := &param.SearchFilter{
```
Comment (rkumpf) [inline examples/filters.go:29]: It might be helpful to show how to specify multiple filters.

File: examples/initiator_group.go [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+
+		// create acl
+		_, err = aclSvc.CreateAccessControlRecord(acl)
+		if err != nil {
+			fmt.Println(err)
+		}
+	}
+
+	// cleaup
+	volume, _ = volSvc.GetVolumeById(*volume.ID)
```
Comment (rkumpf) [inline examples/initiator_group.go]: Is this statement relevant?

File: examples/initiator_group.go [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+		AccessProtocol:  nimbleos.NsAccessProtocolIscsi,
+		IscsiInitiators: initiatorList,
+	}
+
+	// create initiator group
+	igroup, _ = igroupSvc.CreateInitiatorGroup(igroup)
+
+	// create a new volume and assigne acl
+	volume, _ := volSvc.CreateVolume(newVolume)
+	if volume != nil {
```
Comment (rkumpf) [inline examples/initiator_group.go:76]: Is it better to check for volume != nil or for err == nil?

File: examples/restore.go [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+		LimitIops:         &limitIopsField,
+		LimitMbps:         &limitMbpsField,
+		MultiInitiator:    param.NewBool(true),
+		AgentType:         nimbleos.NsAgentTypeNone,
+	}
+
+	// create volume
+	volume, err := volSvc.CreateVolume(newVolume)
+	if volume != nil {
+		volcollname := &nimbleos.VolumeCollection{
```
Comment (rkumpf) [inline examples/restore.go]: Change volcollname to volcoll?

File: examples/restore.go [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+			fmt.Printf("Failed to remove %s volume from volume collection", *volume.ID)
+			return
+		}
+		// delete volcoll
+		groupService.GetVolumeCollectionService().DeleteVolumeCollection(*volcoll.ID)
+		volSvc.DeleteVolume(*volume.ID)
+
+	}
+	// logout
+	groupService.LogoutService()
```
Comment (rkumpf) [inline examples/restore.go]: Should we use a "defer" statement right after initialization so the logout occurs even when we hit the various "return" statements?

File: examples/restore.go [hunk 10]
```diff
@@ -0,0 +1,89 @@
...
+
+	// create volume
+	volume, err := volSvc.CreateVolume(newVolume)
+	if volume != nil {
+		volcollname := &nimbleos.VolumeCollection{
+			Name: param.NewString("TestVolcoll"),
+		}
+		volcoll, err := groupService.GetVolumeCollectionService().CreateVolumeCollection(volcollname)
+		if err != nil {
+			fmt.Println("Failed to get volume collection")
```
Comment (rkumpf) [inline examples/restore.go]: "get" -> "create"

File: pkg/service/volume_service.go [hunk 11]
```diff
@@ -71,6 +72,19 @@ func (svc *VolumeService) GetVolumeById(id string) (*nimbleos.Volume, error) {
 	return volumeResp, nil
 }
 
+// GetVolumeStats - method returns a pointer to "Stats"
+func (svc *VolumeService) GetVolumeStats(id string) (*nimbleos.Stat, error) {
```
Comment (rkumpf) [inline pkg/service/volume_service.go]: How might one provide filters to this call?  For example, "GET /v1/volumes/<id>/stats?sensors=reads,writes".

File: examples/initiator_group.go [hunk 12]
```diff
@@ -0,0 +1,104 @@
...
+		}
+	}
+
+	// cleaup
+	volume, _ = volSvc.GetVolumeById(*volume.ID)
+	updateVol := &nimbleos.Volume{
+		Online: param.NewBool(false),
+	}
+	// set volume offline
+	volume, _ = volSvc.UpdateVolume(*volume.ID, updateVol)
```
Comment (rkumpf) [inline examples/initiator_group.go:94]: Earlier we had talked about providing an example of how a client program could handle a 202 Accepted response when WaitForJobs==false.  Is that planned?

File: examples/fields.go (hunk 2 above)
Comment (vidhutsingh) [reply examples/fields.go:15]: It shd exit

File: examples/fields.go (hunk 3 above)
Comment (vidhutsingh) [reply examples/fields.go:42]: sure

File: examples/fields.go (hunk 4 above)
Comment (vidhutsingh) [reply examples/fields.go:31]: We can use it . lets keep it for now .

File: examples/filters.go (hunk 5 above)
Comment (vidhutsingh) [reply examples/filters.go:29]: we can add a different example, not in this file

File: examples/initiator_group.go (hunk 6 above)
Comment (vidhutsingh) [reply examples/initiator_group.go]: Not required

File: examples/restore.go (hunk 8 above)
Comment (vidhutsingh) [reply examples/restore.go]: No, it should be volcollName

File: examples/restore.go (hunk 9 above)
Comment (vidhutsingh) [reply examples/restore.go]: Make sense .

File: examples/restore.go (hunk 10 above)
Comment (vidhutsingh) [reply examples/restore.go]: ok

File: pkg/service/volume_service.go (hunk 11 above)
Comment (vidhutsingh) [reply pkg/service/volume_service.go]: Filter is not supported for now .

File: examples/initiator_group.go (hunk 12 above)
Comment (vidhutsingh) [reply examples/initiator_group.go:94]: You just need to parse the error code and filter out job id (by using regex ) and invoke GetJobId() . I will create a separate example for it . Not here

File: examples/fields.go (hunk 2 above)
Comment (rkumpf) [reply examples/fields.go:15]: The same comment applies to several of the examples.

File: examples/fields.go (hunk 3 above)
Comment (rkumpf) [reply examples/fields.go:42]: The same comment applies to several of the examples.

**Review by rkumpf — DISMISSED** [review]
It is important that best practices are exemplified, because people will definitely copy and paste from these examples directly into their client programs.

**Review by rkumpf — APPROVED** [review]

**Pull request merged**